apache/beam
# Configure flutter_code_editor options with Hugo shortcode (#23926)

## This PR:
- Resolves #22357
- Resolves #22359
- Resolves #23925
- Resolves #23926 
- Adds `HttpExampleLoader`.
- Uses `null` for lack of complexity instead of a special enum value.

## To test the new features:

1. Add the following code to /website/www/site/content/en/documentation/programming-guide.md:
```
{{< playground height="700px" >}}
{{< playground_snippet language="java" example="SDK_JAVA/PRECOMPILED_OBJECT_TYPE_KATA/AggregationCount" >}}
{{< playground_snippet language="py" example="https://raw.githubusercontent.com/apache/beam/master/sdks/python/apache_beam/examples/snippets/snippets.py" unfold="pipelines_constructing_creating" readonly="pipelines_constructing_writing" >}}
{{< playground_snippet language="go" example="https://raw.githubusercontent.com/apache/beam/master/sdks/go/examples/snippets/01_03intro.go" show="pipeline_options" >}}
{{< playground_snippet language="scio" example="SDK_SCIO/PRECOMPILED_OBJECT_TYPE_EXAMPLE/MinimalWordCount" >}}
{{< /playground >}}
```

2. Modify /website/www/site/layouts/shortcodes/playground.html to replace `https://beam.playground.apache.org` with `http://localhost:...` with your local Playground port.

3. Follow the instructions in https://github.com/apache/beam/tree/master/website to run the website locally.

4. Navigate to http://localhost:1313/documentation/programming-guide/

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:


See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[![Build python source distribution and wheels](https://github.com/apache/beam/workflows/Build%20python%20source%20distribution%20and%20wheels/badge.svg?branch=master&event=schedule)](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[![Python tests](https://github.com/apache/beam/workflows/Python%20tests/badge.svg?branch=master&event=schedule)](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[![Java tests](https://github.com/apache/beam/workflows/Java%20Tests/badge.svg?branch=master&event=schedule)](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[![Go tests](https://github.com/apache/beam/workflows/Go%20tests/badge.svg?branch=master&event=schedule)](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.
